src/app/directives/filedrag.directive.ts

```diff
@@ -59,20 +57,18 @@ export class FiledragDirective {
         //import CodeCoffer xml (from the old desktop app)
         const newSnippets: Snippet[] = this.parser.parse(read.result.toString());
         newSnippets.forEach(snippet => this.service.addSnippet(snippet));
-      }
-      else if (droppedFile.name.toLocaleLowerCase().endsWith('.jcoff')) {
+      } else if (droppedFile.name.toLocaleLowerCase().endsWith('.jcoff')) {
         //import CodeCoffer JSON
         const newSnippets: Snippet[] = JSON.parse(read.result.toString());
         this.service.import(newSnippets);
-      }
-      else {
+      } else {
         //import lone snippet
         const newSnippet: Snippet = new Snippet();
         newSnippet.title = droppedFile.name;
         newSnippet.tags = droppedFile.name;
```

Comment:
You could probably grab the file extension here and then use that to populate the Snippet's language.